### PR TITLE
Decode file URIs

### LINKED ISSUE #23: Unable to work with files with paths that include characters that must be url encoded
Related: https://github.com/Polymer/polymer-editor-service/issues/9

In fact, it seems as though the editor service might handle this correctly, and this was entirely a vscode problem.

### PR DESCRIPTION
**Pull request by rictic**
Fixes #23 

Should improve windows support (https://github.com/Polymer/polymer-editor-service/issues/7) for vscode. I won't be able to test windows until tonight or tomorrow though.

### CODE REVIEW

**Comment by quangld** [comment]
This won't fix the problem on Windows. You still need to remove the first forward slash.
for example: 
while the workspaceRoot = d:\Project\polymer
decodeURI(match[1]) returns = /d:/Project\polymer....

so path.relative won't see these 2 the same path

**Comment by rictic** [comment]
Good catch! I got a basic windows environment setup, and with the latest commit this appears to be working well in windows.

File: server/src/server.ts [hunk 1]
```diff
@@ -178,11 +178,20 @@ connection.onCompletion(async function(
 });
 
 function getWorkspacePathToFile(doc: {uri: string}): string|undefined {
-  const match = doc.uri.match(/^file:\/\/(.*)/);
+  const match = doc.uri.match(/^file:\/\/\/?(.*)/);
```
Comment (justinfagnani) [inline server/src/server.ts]: I'd love comments on regexes to explain what they're trying to match. Here I'd want to highlight the trailing slash, perhaps.

File: server/src/server.ts [hunk 2]
```diff
@@ -178,11 +178,20 @@ connection.onCompletion(async function(
...
     return undefined;
   }
-  return path.relative(workspaceRoot, match[1]);
+  // Decode the URI encoding. TODO: something something unicode?
+  const decodedPath = decodeURIComponent(match[1]);
+  // We've stripped off the leading `/`, which on unix means that we have a
+  // relative-looking path, but on Windows we still have an absolute path that
+  // starts with e.g. `C:\`. We can normalize these two by forcing both into
+  // an absolute path, like so:
+  const absolutePath = path.resolve('/', decodedPath);
```
Comment (justinfagnani) [inline server/src/server.ts]: Tests would be cool, but you probably want to set up Appveyor before they would catch anything important from this change.

**Comment by LostInBrittany** [comment]
After having installed the [0.0.2 release](https://github.com/Polymer/vscode-plugin/releases/tag/v0.0.2) I still have see the  `c:\c:\` path

**Comment by LostInBrittany** [comment]
Oh, it seems this PR hasn't made it into the 0.0.2 release...

**Pull request merged**